sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -470,9 +470,15 @@ trait CheckAnalysis extends PredicateHelper {
               }
             }
 
+            val colsToDelete = mutable.Set.empty[Seq[String]]
```

Comment:
This causes conflicts when I backport https://github.com/apache/spark/pull/27584

I think the change in this file should go into 3.0 as well. Logically columns deleted should be skipped when checking name duplication for AddColumn.

@imback82 can you open a PR to backport https://github.com/apache/spark/pull/27584 with changes in this file?

Reply:
Yes, working on it now!